src/components/header.js

```diff
@@ -11,39 +14,130 @@ const menus = [
   { label: "Jobs", path: "/jobs" },
 ]
 
-const Header = ({ siteTitle }) => (
-  <Box
-    as="header"
-    sx={{
-      backgroundColor: "transparent",
-      position: "absolute",
-      top: 0,
-      left: 0,
-      right: 0,
-      zIndex: 999,
-    }}
-  >
-    <Container>
-      <Flex
-        color="white"
-        justifyContent="space-between"
-        alignItems="center"
-        py={2}
-      >
-        <GatsbyLink to="/">
-          <Image src={Logo} alt={siteTitle} width="12.5rem" />
-        </GatsbyLink>
-        <Box>
-          {menus.map(menu => (
-            <Link key={menu.label} variant="nav" href={menu.path}>
-              {menu.label}
-            </Link>
-          ))}
-        </Box>
-      </Flex>
-    </Container>
-  </Box>
-)
+const Header = ({ siteTitle }) => {
+  const size = useWindowSize()
+  const [isMobile, setIsMobile] = useState()
+  const [showMenu, setShowMenu] = useState()
+
+  useEffect(() => {
+    if (size.width < 640) {
+      setIsMobile(true)
+      setShowMenu(false)
+    } else {
+      setIsMobile(false)
+      setShowMenu(false)
+    }
+  }, [size])
+
+  useEffect(() => {
+    if (showMenu) {
+      document.body.style.overflow = "hidden"
+    } else {
+      document.body.style.overflow = "visible"
+    }
+  }, [showMenu])
+
+  const handleClickOutsideNav = e => {
```

Comment:
I would love it if you just convert this to a hook. but not really required :squirrel: